gmp-droid.cpp

```diff
@@ -188,20 +188,25 @@ class DroidVideoDecoder : public GMPVideoDecoder
             return;
         }
 
-        // Check that we won't run out of space in the buffer
-        if (offset + len + 4 > inputFrame->Size ()) {
-          LOG (ERROR,
-              "Not enough space left for " << len << " bytes of NAL data");
-          Error (GMPDecodeErr);
-          return;
-        }
-        // Write NAL start code over the lenth
-        static const uint8_t code[] = { 0x00, 0x00, 0x00, 0x01 };
-        const uint8_t *start_code = code + (4 - start_code_len);
-        memcpy (inputFrame->Buffer () + offset, start_code, start_code_len);
-        offset += start_code_len + len;
 
-        LOG (DEBUG, "Parsed nal unit of size " << len);
+        if (len == 1) {
```

Comment:
ESR60 seems to have processed this already. OpenH264 sdoesn't allow for this, so I'm not sure why it doesn't need this. Perhaps later versions of Gecko stopped doing it again.

Reply:
ESR60 has special handling for openh264 to supply it with the same AVCC format as before. It's changed to Annex B for everything else.